psychopy/visual/vlcmoviestim.py

```diff
@@ -0,0 +1,625 @@
+            self._retracerate = 60.0
+        self.filename = pathToString(filename)
+        self.loop = loop
+        self.flipVert = flipVert
+        self.flipHoriz = flipHoriz
+        self.pos = numpy.asarray(pos, float)
+        self.size = numpy.asarray(size, float)
+        self.depth = depth
+        self.opacity = float(opacity)
+        self.volume = volume
+        self.no_audio = noAudio
+
+        self.interpolate = interpolate
+        self._texture_id = GL.GLuint()
+        GL.glGenTextures(1, ctypes.byref(self._texture_id))
+
+        self._pause_time = 0
+        self._vlc_clock = Clock()
+        self._vlc_initialized = False
+        self._reset()
+        self.loadMovie(self.filename)
+        self.setVolume(volume)
+        self.nDroppedFrames = 0
+
+        self.ori = ori
+        # set autoLog (now that params have been initialised)
+        self.autoLog = autoLog
+        if autoLog:
+            logging.exp("Created {} = {}".format(self.name, self))
+
+    def _reset(self):
+        self.frame_counter = 0
+        self.current_frame = 0
+        self.duration = None
+        self.status = NOT_STARTED
+        self.width = None
+        self.height = None
+        self.frame_rate = None
+
+        if self._vlc_initialized:
+            self._release_vlc()
+
+    def setMovie(self, filename, log=True):
+        """See `~MovieStim.loadMovie` (the functions are identical).
+
+        This form is provided for syntactic consistency with other
+        visual stimuli.
+        """
+        self.loadMovie(filename, log=log)
+
+    def loadMovie(self, filename, log=True):
+        """Load a movie from file
+
+        :Parameters:
+
+            filename: string
+                The name of the file, including path if necessary
+
+        Due to VLC oddness, .duration is not correct until the movie starts playing.
+        """
+        self._reset()
+        self.filename = pathToString(filename)
+
+        # Initialize VLC
+        self._vlc_start()
+
+        self.status = NOT_STARTED
+        logAttrib(self, log, 'movie', filename)
+
+    def _vlc_start(self):
+        """
+        Create the vlc stream player for the video using python-vlc.
+        """
+        if not os.access(self.filename, os.R_OK):
+            raise RuntimeError('Error: %s file not readable' % self.filename)
+        if self.no_audio:
+            instance = vlc.Instance("--no-audio")
+        else:
+            instance = vlc.Instance()
+        try:
+            stream = instance.media_new(self.filename)
+        except NameError:
+            msg = 'NameError: %s vs LibVLC %s'
+            raise ImportError(msg % (vlc.__version__,
+                                     vlc.libvlc_get_version()))
+
+        player = instance.media_player_new()
+        player.set_media(stream)
+
+        # Load up the file
+        stream.parse()
+        size = player.video_get_size()
+        self.video_width = size[0]
```

Comment:
Attributes of PsychPy classes should always be camelCase rather than snake_case
(see [PEP8 special cases](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds) for explanation)